internal/services/maintenance/maintenance_assignment_dedicated_host_resource.go

```diff
@@ -64,41 +66,45 @@ func resourceArmMaintenanceAssignmentDedicatedHostCreate(d *pluginsdk.ResourceDa
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	dedicatedHostIdRaw := d.Get("dedicated_host_id").(string)
-	dedicatedHostId, _ := parseCompute.DedicatedHostID(dedicatedHostIdRaw)
+	dedicatedHostId, err := parseCompute.DedicatedHostID(d.Get("dedicated_host_id").(string))
+	if err != nil {
+		return err
+	}
 
-	existingList, err := getMaintenanceAssignmentDedicatedHost(ctx, client, dedicatedHostId, dedicatedHostIdRaw)
+	existingList, err := getMaintenanceAssignmentDedicatedHost(ctx, client, dedicatedHostId, dedicatedHostId.ID())
 	if err != nil {
 		return err
 	}
 	if existingList != nil && len(*existingList) > 0 {
 		existing := (*existingList)[0]
-		if existing.ID != nil && *existing.ID != "" {
-			return tf.ImportAsExistsError("azurerm_maintenance_assignment_dedicated_host", *existing.ID)
+		if existing.Id != nil && *existing.Id != "" {
+			return tf.ImportAsExistsError("azurerm_maintenance_assignment_dedicated_host", *existing.Id)
```

Comment:
rather than using the API's Resource ID could we use the Terraform Resource ID that we're building up?